### DISCUSSION TITLE
How to find the TERMINFO supplied by ncurses port

### DISCUSSION THREAD

**Discussion by pbishop62** [zopencommunity/meta#658] [Q&A] [answered]
Hi,

please excuse this somewhat naive question, as I'm quite new to the z/OS Open Tools ports, only having installed them today for the first time.

I am using PuTTY on z/OS 2.5.  I installed ncurses with "zopen install ncurses".

```
PBISHOP:/u/pbishop/testing/meta-0.8.1: >zopen install ncurses
- Querying repo for latest package information.
Processing package: ncurses
- Installing ncurses...
- Downloading ncurses-6.4.20231114_180107.zos.pax.Z file from remote to zopen package cache...
- Download complete.
- Processing ncurses-6.4.20231114_180107.zos.pax.Z...
After this operation, 22.99 MB of additional disk space will be used.
Do you want to continue? [y/n]
y
- Expanding ncurses-6.4.20231114_180107.zos.pax.Z
- Expanded
- Integration complete.
- Checking for env file.
- .env file found, adding to profiled processing.
- Sourcing environment to run any setup.
Setting up ncurses...
Setup completed.
- Checking for runtime dependencies.
- No runtime dependencies found.
Successfully installed: ncurses
```

I can run infocmp -V as suggested in the test.sh shown at https://github.com/ZOSOpenTools/ncursesport/blob/main/test.sh

```
PBISHOP:/u/pbishop/testing: >infocmp -V
ncurses 6.4.20221231
```

However when I try tput, I get the following:

```
PBISHOP:/u/pbishop/testing: >tput
tput: unknown terminal "linux"
```

I have exported TERMINFO to point to the supplied terminfo directory which has heaps of defined terminals (much more than the defaults that come with Unix System Services), including the 'linux' entry in the 'l' subdirectory.

`TERMINFO="/u/pbishop/zopen/usr/local/share/terminfo/"
`

So my question is, how do I get the supplied TERMINFO database to be used by the zopen ncurses port?

I admit to being a novice, so be gentle  :-)

I did see this earlier issue, but couldn't quite see how it helps me:  https://github.com/orgs/ZOSOpenTools/discussions/195

I tried this, but it didn't seem to work:

```
PBISHOP:/u/pbishop/testing: >. $ZOPEN_PKGINSTALL/ncurses/ncurses-6.4.20231114_180107.zos/.env
Need to source from the .env directory
```

This also failed, but I was unwilling to make it executable as that might break something else:

```
PBISHOP:/u/pbishop/testing: >/u/pbishop/zopen/etc/profiled/ncurses/dotenv
/u/pbishop/zopen/etc/profiled/ncurses/dotenv: FSUM9209 cannot execute: reason code = ef076015: EDC5111I Permission denied.
```

Thanks in advance for any help.  I'll keep digging around to see if anything else presents the answer.

I am looking forward to getting more familiar with this project as it seems very inspiring.

Kind regards,
Peter Bishop

**Comment by MikeFultonDev** [comment, marked as answer]
I am not sure if this is the issue but I
`export TERM=xterm`

**Reply by pbishop62** [reply]
Thanks Mike, much obliged.  I am now getting sensible replies from tput, e.g.

```
PBISHOP:/u/pbishop: >tput
Usage: tput [options] [command]

Options:
  -S <<       read commands from standard input
  -T TERM     use this instead of $TERM
  -V          print curses-version
  -x          do not try to clear scrollback

Commands:
  clear       clear the screen
  init        initialize the terminal
  reset       reinitialize the terminal
  capname     unlike clear/init/reset, print value for capability "capname"
PBISHOP:/u/pbishop: >tput -V
ncurses 6.4.20221231
```

Best regards, onward we go.

Peter Bishop

**Reply by IgorTodorovskiIBM** [reply]
One other point, the z/OS Open Tools ncurses uses ASCII_TERMINFO for its database. It should be set automatically when you source the zopen-config.
```
. $ZOPEN_ROOTFS/etc/zopen-config
env | grep ASCII_TERMINFO
```

**Comment by pbishop62** [comment]
Yep, that variable is set as expected.  Ncurses now works as expected with my minimalist test program.

Thanks heaps!

cheers,
Peter
